src/plugins/cachefilter/cachefilter.c

```diff
@@ -42,7 +42,7 @@ int elektraCachefilterGet (Plugin * handle, KeySet * returned, Key * parentKey)
 	void * cache = elektraPluginGetData (handle);
 	if (cache == NULL)
 	{
-		toBeCached = ksNew (ksGetSize (returned), KS_END);
+		toBeCached = ksNew (0, KS_END);
```

Comment:
Did you benchmark this? It might waste some space but should avoid many reallocations (for larger keysets).